test/test_user_pref_sched.py

```diff
@@ -524,6 +524,34 @@ def test_task_5(self):
         with patch.object(data_access, 'data_access_levels', patch_data_access_levels):
             assert n_available_tasks_for_user(project, 501) == 3
 
+    @with_context
```

Comment:
Can you set a more descriptive name for this test to be clear on what are we testing.  Since there are multiple use cases will be good to be clear which one are you testing.